contracts/ProtocolReserve/ProtocolShareReserve.sol

```diff
@@ -328,8 +344,8 @@ contract ProtocolShareReserve is AccessControlledV8, ReentrancyGuardUpgradeable,
         uint256 totalSchemas = uint256(type(Schema).max) + 1;
         uint256[] memory schemaBalances = new uint256[](totalSchemas);
         uint256 totalBalance;
-
-        for (uint256 schemaValue; schemaValue < totalSchemas;) {
+        _ensureMaxLoops(totalSchemas);
```

Comment:
I think the check should be done when a new schema is added, not here. But we don't have a function to add new schemas, so maybe it's enough if we remove this check